activerecord/test/cases/migration/pending_migrations_test.rb

```diff
@@ -92,25 +87,24 @@ def test_with_stdlib_logger
         end
 
         private
-          def assert_pending_migrations
-            # Do twice to test that the error continues to be raised.
-            2.times do
```

Comment:
Running this 2 times actually demonstrates the bug I was seeing - we should not need to run it twice, it should find all the pending migrations the first time.

Reply:
My impression was that running it twice was not to detect more pending migrations, but rather to ensure that detecting a pending migration does not alter the migration's status: https://github.com/rails/rails/blob/7eff9d6f683239140303dd57af97e06d899309b5~1/activerecord/test/cases/migration/pending_migrations_test.rb#L46-L53

But, as you say, that might be unnecessary.